operator/internal/manifests/networkpolicy.go

```diff
@@ -0,0 +1,596 @@
+				// Allow egress to object storage
+				{
+					To: []networkingv1.NetworkPolicyPeer{},
+					Ports: []networkingv1.NetworkPolicyPort{
+						// Most S3 endpoints AWS S3, GCS, Azure, ODF
+						{
+							Protocol: ptr.To(corev1.ProtocolTCP),
+							Port:     &intstr.IntOrString{Type: intstr.Int, IntVal: 443},
+						},
+						// MinIO
+						{
+							Protocol: ptr.To(corev1.ProtocolTCP),
+							Port:     &intstr.IntOrString{Type: intstr.Int, IntVal: 9000},
+						},
+						// Alternative popular port for S3 endpoints
+						{
+							Protocol: ptr.To(corev1.ProtocolTCP),
+							Port:     &intstr.IntOrString{Type: intstr.Int, IntVal: 8080},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+// buildGatewayAllow NetworkPolicy to allow ingress and egress traffic from
+// gateway pods
+func buildGatewayAllow(opts Options) *networkingv1.NetworkPolicy {
+	promNetworkPolicyPeer := networkingv1.NetworkPolicyPeer{}
+	if opts.Stack.Tenants.Mode == lokiv1.OpenshiftLogging || opts.Stack.Tenants.Mode == lokiv1.OpenshiftNetwork {
+		// Running in OpenShift so we can restrict to only the Prometheus pods in
+		// the openshift-monitoring
+		promNetworkPolicyPeer = networkPolicyPeerPrometheusPods
+	}
+
+	return &networkingv1.NetworkPolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-gateway-allow", opts.Name),
+			Namespace: opts.Namespace,
+			Labels:    commonLabels(opts.Name),
+		},
+		Spec: networkingv1.NetworkPolicySpec{
+			PodSelector: *selectorLokiGatewayPods,
+			PolicyTypes: []networkingv1.PolicyType{
+				networkingv1.PolicyTypeEgress,
+				networkingv1.PolicyTypeIngress,
+			},
+			Egress: []networkingv1.NetworkPolicyEgressRule{
+				egressToDNSK8s,
+				egressToDNSOpenshift,
+				// Allow egress to query-frontend for queries
+				// Allow egress to distributor for pushing logs
+				// Allow egress to ruler for getting rules
+				{
+					To: []networkingv1.NetworkPolicyPeer{
+						{
+							NamespaceSelector: &metav1.LabelSelector{
+								MatchLabels: map[string]string{
+									"kubernetes.io/metadata.name": opts.Namespace,
+								},
+							},
+							PodSelector: &metav1.LabelSelector{
+								MatchExpressions: []metav1.LabelSelectorRequirement{
+									{
+										Key:      "app.kubernetes.io/name",
+										Operator: metav1.LabelSelectorOpIn,
+										Values:   []string{"lokistack"},
+									},
+									{
+										Key:      "app.kubernetes.io/component",
+										Operator: metav1.LabelSelectorOpIn,
+										Values:   []string{"distributor", "query-frontend", "ruler"},
+									},
+								},
+							},
+						},
+					},
+					Ports: []networkingv1.NetworkPolicyPort{
+						httpPolicyPort,
+					},
+				},
+				// Allow egress to the API server for token requests
+				{
+					To: []networkingv1.NetworkPolicyPeer{},
+					Ports: []networkingv1.NetworkPolicyPort{
+						{
+							Protocol: ptr.To(corev1.ProtocolTCP),
+							Port:     &intstr.IntOrString{Type: intstr.Int, IntVal: 6443},
```

Comment:
Interesting. I would have guessed that we need the service port here and not the container port (`kubernetes.default.svc` has port 443 in my testcluster).

Is it maybe possible to specify "egress to the `kubernetes.default.svc`" instead of just listing the port here?

Reply:
This one did it like so because of https://docs.google.com/document/d/1CDoGSRd-h8VT4PMrK_83Ro0YzYPjORbkxtfTJU1sN6Q/edit?tab=t.0#heading=h.r5za8itrk0hm